xclim/indices/_hydrology.py

```diff
@@ -279,3 +283,120 @@ def melt_and_precip_max(
+    30th September, as commonly defined in North America.
+
+    Parameters
+    ----------
+    q : xr.DataArray
+        Daily streamflow data.
+    threshold_factor : int
+        Factor by which the median flow is multiplied to set the high flow threshold, default is 9.
+    freq : str, optional
+        Resampling frequency, default is 'A-SEP' for water year ending in September.
+    indexer
+        Indexing parameters to perform a temporal subset of the data.
+        It accepts the same arguments as :py:func:`xclim.indices.generic.select_time`.
+
+    Returns
+    -------
+    xr.DataArray
+        Calculated mean of high flow days per water year
+
+    References
+    ----------
+    1. Addor, Nans & Nearing, Grey & Prieto, Cristina & Newman, A. & Le Vine, Nataliya & Clark, Martyn. (2018). A Ranking of Hydrological Signatures Based on Their Predictability in Space. Water Resources Research. 10.1029/2018WR022606.
+    2. Clausen, B., & Biggs, B. J. F. (2000). Flow variables for ecological studies in temperate streams: Groupings based on covariance. Journal of Hydrology, 237(3–4), 184–197. https://doi.org/10.1016/S0022-1694(00)00306-1
+    """
+    median_flow = q.median(dim="time")
+    with xr.set_options(keep_attrs=True):
+        threshold = threshold_factor * median_flow
+
+    sel = select_time(q, **indexer)
+    out = threshold_count(sel, ">", threshold, freq=freq)
+    return to_agg_units(out, q, "count")
+
+
+@declare_units(q="[discharge]")
+def low_flow_frequency(
+    q: xr.DataArray, threshold_factor: float = 0.2, freq: str = "A-SEP", **indexer
+) -> xr.DataArray:
+    """
+    Calculate the number of days in a given period with flows lower than a specified threshold, given by a fraction
+    of the mean flow. By default, the period is the water year starting on 1st October and ending on 30th September,
+    as commonly defined in North America.
+
+    Parameters
+    ----------
+    q : xr.DataArray
+        Daily streamflow data.
+    threshold_factor : float
+        Factor by which the mean flow is multiplied to set the low flow threshold, default is 0.2.
+    freq : str, optional
+        Resampling frequency, default is 'A-SEP' for water year ending in September.
+    indexer
+        Indexing parameters to perform a temporal subset of the data.
+        It accepts the same arguments as :py:func:`xclim.indices.generic.select_time`.
+
+    Returns
+    -------
+    xr.DataArray
+        Calculated mean of low flow days per water year.
+
+    References
+    ----------
+    Olden, J. D., & Poff, N. L. (2003). Redundancy and the choice of hydrologic indices for characterizing streamflow regimes. River Research and
+    Applications, 19(2), 101–121. https://doi.org/10.1002/rra.700
+    """
+    mean_flow = q.mean(dim="time")
+    with xr.set_options(keep_attrs=True):
+        threshold = threshold_factor * mean_flow
```

Comment:
```suggestion
    threshold = threshold_factor * mean_flow
```
Why did you need `keep_attrs=True` here ? Usually, indices drop attributes. Indicators preserve them, but explicitly in the `Indicator` class.